ioctl/test_ioctl.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    Phoenix-RTOS standard library functions tests
+ *    HEADER:
+ *    - ioctl.h
+ *    TESTED:
+ *    - ioctl()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <phoenix/ioctl.h>
+#include <sys/msg.h>
+#include <signal.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <posix/utils.h>
+#include <unity_fixture.h>
+
+
+static uint32_t port;
+static pthread_t tid;
+static int fd, dd;
+
+#define BIG_SIZE 1024
+#define MAX_FAIL 10
+#define PATH     "test_ioctl_tmpFile"
+/* In above T stands for test, do not confuse with tty */
+#define TIOCSIG  _IOC(0, 'T', 0x01, 0)
+#define TIOCVAL  _IOC(0, 'T', 0x02, 4)
+#define TIOCPTR  _IOC(IOC_IN, 'T', 0x03, 4)
+#define TIOCPTR2 _IOC(IOC_IN, 'T', 0x04, BIG_SIZE)
+#define TIOCOUT  _IOC(IOC_OUT, 'T', 0x05, 4)
+#define TIOCOUT2 _IOC(IOC_OUT, 'T', 0x06, BIG_SIZE)
```

Comment:
Some `inout` could be valuable, so both passing some value and reading the another one.